Platform support for the DELL S6010 with the Rangeley C2538 CPU module

diff --git a/drivers/platform/x86/Kconfig b/drivers/platform/x86/Kconfig
index 7f43cf8..b1acf68 100644
--- a/drivers/platform/x86/Kconfig
+++ b/drivers/platform/x86/Kconfig
@@ -779,4 +779,17 @@ config SAMSUNG_Q10
 	  This driver provides support for backlight control on Samsung Q10
 	  and related laptops, including Dell Latitude X200.
 
+config DELL_S6010_C2538
+	tristate "DELL S6010_C2538 Platform Support"
+	depends on X86 && PCI
+	select I2C_ISCH
+	select I2C_ISMT
+	select GPIO_SCH
+	select MFD_CORE
+	select LPC_SCH
+	---help---
+	 Enable support for DELL S6010 switching platform.  This
+	 platform has a C2538 Atom (Rangeley) CPU module, a Broadcom
+	 BCM56860 switch ASIC. It has 32x40G QSFP+ ports.
+
 endif # X86_PLATFORM_DEVICES
